MRN-Code/coinstac
# Ensure computations’ stdout JSON errors are human-readable

### Problem

Errors in parsing computations’ stdout as JSON aren’t clearly visible to authors (see #49)
### Solution

Use [json-parse-helpfulerror](https://www.npmjs.com/package/json-parse-helpfulerror). Clean up the method and tests. Bump coverage to actual 100%.
### Testing
1. Check out the _enhancement/#49-command-json-error_ branch in your local repo
2. Run `npm run bootstrap` from the repo’s root to install new dependencies
3. Run `npm link` in _packages/coinstac-simulator_ to globally link the package
4. Create the following files in a test directory:
   
   ``` js
   // computation.js
   module.exports = {
     name: 'my-bad-thingy',
     version: '1.0.0',
     local: {
       type: 'cmd',
       cmd: 'node',
       args: ['./hello-dude.js'],
       verbose: true,
     },
     remote: {
       type: 'fn',
       fn: function(params) {
         console.log(params);
         return { complete: true };
       },
     },
   };
   ```
   
   ``` js
   // declaration.js
   module.exports = {
     computationPath: './computation.js',
     local: Array(1),
     verbose: true,
   };
   ```
   
   ``` js
   // hello-dude.js
   console.log(`[{
     "hello": "dude",
     "what": "is up"
   }, {
     "not": "much"
   }, {
     "yolo": 88,
   }]`);
   ```
5. Run `coinstac-simulator -d declaration.js` in your test directory
6. Observe easy-to-understand JSON parsing error

Closes #49.
